src/fheroes2/dialog/dialog_selectitems.cpp

```diff
@@ -1000,10 +972,172 @@ int Dialog::selectHeroes( const int heroId /* = Heroes::UNKNOWN */ )
+    raceSpriteBorder.reset();
+    const int32_t raceImageWidth = raceBorderWidth - 8;
+    const int32_t raceImageHeight = raceBorderHeight - 8;
+    fheroes2::Copy( randomRaceSprite, 0, 0, raceSpriteBorder, 0, 0, raceBorderWidth, 4 );
+    fheroes2::Copy( randomRaceSprite, 0, 4, raceSpriteBorder, 0, 4, 4, raceImageHeight );
+    int32_t imageCopyOffset = raceBorderWidth - 4;
+    fheroes2::Copy( randomRaceSprite, imageCopyOffset, 4, raceSpriteBorder, imageCopyOffset, 4, 4, raceImageHeight );
+    imageCopyOffset = raceBorderHeight - 4;
+    fheroes2::Copy( randomRaceSprite, 0, imageCopyOffset, raceSpriteBorder, 0, imageCopyOffset, raceBorderWidth, 4 );
+    fheroes2::Image raceSpriteBorderSelected( raceSpriteBorder );
+    fheroes2::ApplyPalette( raceSpriteBorder, darkGrayPalette );
+    fheroes2::ApplyPalette( raceSpriteBorderSelected, PAL::GetPalette( PAL::PaletteType::RED ) );
+
+    std::array<fheroes2::Rect, 6> colorRect;
+    std::array<fheroes2::Rect, 7> raceRect;
+
+    // Random race.
+    raceRect.back() = { pos.x - 2, pos.y + raceOffsetY, raceBorderWidth, raceBorderHeight };
+    fheroes2::Copy( randomRaceSprite, 4, 4, display, raceRect.back().x + 4, raceRect.back().y + 4, raceImageWidth, raceImageHeight );
+    fheroes2::Blit( raceShadow, display, pos.x - 5, pos.y + raceOffsetY + 2 );
+    text.set( _( "race|Random" ), fheroes2::FontType::smallWhite() );
+    text.draw( raceRect.back().x, raceRect.back().y + raceRect.back().height + 4, raceBorderWidth, display );
+
+    pos.x += stepX;
+    for ( uint32_t i = 0; i < 6; ++i ) {
+        const fheroes2::Sprite & colorSprite = fheroes2::AGG::GetICN( ICN::BRCREST, i );
+
+        fheroes2::addGradientShadow( colorSpriteBorder, display, { pos.x - stepX / 2, pos.y }, { -5, 5 } );
+        colorRect[i] = { pos.x - stepX / 2, pos.y, colorSpriteBorder.width(), colorSpriteBorder.height() };
+        fheroes2::Copy( colorSprite, 0, 0, display, colorRect[i].x + 4, colorRect[i].y + 4, colorSprite.width(), colorSprite.height() );
+        text.set( Color::String( Color::IndexToColor( static_cast<int>( i ) ) ), fheroes2::FontType::smallWhite() );
+        text.draw( colorRect[i].x + ( colorRect[i].width - text.width() ) / 2, colorRect[i].y + colorRect[i].height + 4, display );
+
+        const fheroes2::Sprite & raceSprite = fheroes2::AGG::GetICN( ICN::NGEXTRA, 51U + i );
+        raceRect[i] = { pos.x - 2, pos.y + raceOffsetY, raceBorderWidth, raceBorderHeight };
+        fheroes2::Copy( raceSprite, 4, 4, display, raceRect[i].x + 4, raceRect[i].y + 4, raceImageWidth, raceImageHeight );
+        fheroes2::Blit( raceShadow, display, pos.x - 5, pos.y + raceOffsetY + 2 );
+        text.set( Race::DoubleLinedString( Race::IndexToRace( static_cast<int>( i ) ) ), fheroes2::FontType::smallWhite() );
+        text.draw( raceRect[i].x, raceRect[i].y + raceRect[i].height + 4, raceBorderWidth, display );
+
+        pos.x += stepX;
+    }
+
+    pos.x = area.x + area.width / 2;
+    pos.y += 170;
```

Comment:
What about moving hero sprite a little lower to avoid possible overlap with two-lined Sorceress translation?
```suggestion
    pos.y += 175;
```
![изображение](https://github.com/ihhub/fheroes2/assets/113276641/78e70aa7-87e9-4b78-8880-7c5ddcea5c1e)